app/src/main/java/com/stonks/calculations/Prediction.kt

```diff
@@ -0,0 +1,33 @@
+package com.stonks.calculations
+
+import org.nield.kotlinstatistics.SimpleRegression
+import org.nield.kotlinstatistics.simpleRegression
+
+class Prediction {
+    class AmountDate (val date: Long, val amount: Double)
+
+    fun prediction (date_array: Array<Long>, amount_array: Array<Double>): SimpleRegression {
+
+        val amountDates = mutableListOf<AmountDate>()
+        for (i: Int in 0..(date_array.size-1)){
+            amountDates.add(AmountDate(date_array[i], amount_array[i]))
+        }
+
+        return amountDates.simpleRegression(
+                xSelector = { it.date },
+                ySelector = { it.amount }
+        )
+    }
+
+    /**
+     * Использование в следующем формате:
+     * (Prediction::datePrediction)(Prediction(),dateTest,amountTest,30)
+     * Где
+     * dateTest: Array<Long>
+     * amountTest: Array<Double>
+     * Возвращает значение типа double через <30> дней
+     */
+    fun datePrediction (date_array: Array<Long>, amount_array: Array<Double>, days: Int): Double {
+        return prediction(date_array,amount_array).predict(date_array[date_array.size-1].toDouble()+days)
```

Comment:
Shouldn't this function return prediction for a period of time? Otherwise we'll have to call it, like, 30 times in our code for a month. That doesn't seem correct. Or maybe I didn't get it right?

Reply:
This is linear regression function, so, while we have first value and last value we can fit all other points, so it isn't necessary to call it 30 times.

Reply:
Reasonable, resolved